### PR TITLE
[FIXED MIRROR] Tsu's Brand Spanking New Storage: or, How I Learned to Refactor For Skyrat

### PR DESCRIPTION
**Pull request by azrael-x86**
## About The Pull Request

**Uber cool.**

image

Currently, storage works as a subtype of `/datum/component`, utilizing `GetComponent()` and signals to operate. While this is a pretty good idea in theory, the execution was pretty trash, and we end up with alot of `GetComponent()` snowflake code (something that shouldn't even need to be used frankly), and a heaping load of scattered procs that lead into one another, and procs that don't get utilized properly.

Instead, this PR adds `atom_storage` and `proc/create_storage(. . .)` to every atom, allowing for the possibility of storage on quite frankly anything. Not only does this entirely remove the need for signals, but it heavily squashes down the number of needed procs in total (removing snowflake signal procs that just lead to one another), reducing overall proc overhead and improving performance.

closes #14817

## Why It's Good For The Game

Reduces proc overhead in total, removes snowflake `GetComponent()` code, removes the need for signals in storage. Datumizes storage.

## Changelog

:cl:
refactor: changes storage from a component to a normal datum
/:cl:

### CODE REVIEW

**Comment by azrael-x86** [comment]
on hold until storage is resolved upstream

File: code/game/objects/items/storage/fancy.dm [hunk 1]
```diff
@@ -21,16 +21,22 @@
...
+	. = ..()
+
+	atom_storage.max_slots = spawn_count
+
 /obj/item/storage/fancy/PopulateContents()
 	if(!spawn_type)
 		return
-	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
-	for(var/i = 1 to STR.max_items)
+	for(var/i = 1 to spawn_count)
```
Comment (GoldenAlpharex) [inline code/game/objects/items/storage/fancy.dm:39]: Could've done `for(var/i in 1 to spawn_count)` instead.

File: code/game/objects/items/storage/wallets.dm [hunk 2]
```diff
@@ -5,18 +5,15 @@
...
 	var/cached_flat_icon
 
-/obj/item/storage/wallet/ComponentInitialize()
+/obj/item/storage/wallet/Initialize()
 	. = ..()
-	var/datum/component/storage/STR = GetComponent(/datum/component/storage/concrete/wallet)
-	STR.max_items = 5
-	STR.set_holdable(list(
-		/obj/item/reagent_containers/hypospray/medipen,
+	atom_storage.max_slots = 4
```
Comment (GoldenAlpharex) [inline code/game/objects/items/storage/wallets.dm:15]: Smh, balance changes.

File: config/admins.txt [hunk 3]
```diff
@@ -137,6 +137,7 @@ shizcalev = Game Master
 NicBR = Game Master
 LoserWasTaken = Game Master
 Fikou = Game Master
+Magatsuchi = Game Master
```
Comment (GoldenAlpharex) [inline config/admins.txt:140]: I really need to clean this file down here, eventually.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me, might have to do a follow-up PR because there's likely one thing somewhere that won't work as expected anymore.

**Pull request merged**